src/unix.rs

```diff
@@ -399,6 +400,80 @@ impl ThreadPriority {
     }
 }
 
+// On apple targets, only `pthread_setschedparam` is allowed. `nice` will fail when increasing the priority if the use is not superuser.
+#[cfg(any(target_os = "macos", target_os = "ios"))]
+impl ThreadPriority {
+    /// Returns the maximum allowed value for using with the provided policy.
+    /// The returned number is in the range of allowed values.
+    pub fn max_value_for_policy(policy: ThreadSchedulePolicy) -> Result<libc::c_int, Error> {
+        let max_priority = unsafe { libc::sched_get_priority_max(policy.to_posix()) };
+        if max_priority < 0 {
+            Err(Error::OS(errno()))
+        } else {
+            Ok(max_priority)
+        }
+    }
+
+    /// Returns the minimum allowed value for using with the provided policy.
+    /// The returned number is in the range of allowed values.
+    pub fn min_value_for_policy(policy: ThreadSchedulePolicy) -> Result<libc::c_int, Error> {
```

Comment:
The relevant code paths changed here would be better to just do right in the original implementation, considering the number of changes required for macOS and iOS.